### PR TITLE
Finished exercises

### PR DESCRIPTION
**Pull request by JohnBain**

### CODE REVIEW

File: answers.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+}
+
+//printPositives([1,2,3,-4])
+
+/* Exercise 2
+
+Similar to the previous exercise, write a function called getPositives that takes an array and uses its filter method to return a 
+new array with only the positive numbers. */
+
+function getPositives (array) {
```
Comment (ziad-saab) [inline answers.js]: @JohnBain the whole point of `filter` is to not have to create an array manually. Can you please review some filtering examples and adapt to this exercise? Come see us if unclear.

File: answers.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+/* Exercise 3
+
+Write a function called filterArray that takes an array AND a function as arguments. 
+Your filter function should return a new array that contains only the elements where the 
+passed function returns a truthy value.
+
+NOTE: You are allowed to use Array.prototype.filter to answer this question. */
+
+function isitTruthy (num) {
```
Comment (ziad-saab) [inline answers.js]: @JohnBain while this is valid, can you see how your function can be much shorter? Think of the general pattern of `if (something is true) {return true;} else {return false;}`

File: answers.js [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+*/
+
+function countVowels(string){
+var count = 0;
+var array = string.split("");
+var vowels = ["A","E", "I", "O", "U", "a", "e", "i", "o", "u"];
+
+return array.reduce(function(count, curr) {
+    if (vowels.indexOf(curr) === -1) {
+        null;
```
Comment (ziad-saab) [inline answers.js]: @JohnBain while this is valid, normally when we have nothing to put in an `if` we simply inverse it. In this case, `if (... > -1)` then you can put the code of your else directly in the if

Comment (JohnBain) [inline answers.js]: Would 

>   if (!vowels.indexOf(curr) === -1) 

work?

File: answers.js (hunk 2 above)
Comment (JohnBain) [inline answers.js]: It could be a one liner, right?

>if (num) {return true;} else {return false;}

File: answers.js (hunk 1 above)
Comment (JohnBain) [inline answers.js]: >function isPositive(value) {
>    return value > 0;
>}

>console.log(([1,2,-3,-4,-5]).filter(isPositive));

Better? I could wrap my filter function in a larger "getPositive" function, but it seems like it'd be extraneous at this point.

File: answers.js (hunk 3 above)
Comment (ziad-saab) [inline answers.js]: @JohnBain yes!!

File: answers.js (hunk 2 above)
Comment (ziad-saab) [inline answers.js]: @JohnBain even shorter ;)

File: answers.js (hunk 1 above)
Comment (ziad-saab) [inline answers.js]: @JohnBain Absolutely correct.

File: answers.js (hunk 2 above)
Comment (JohnBain) [inline answers.js]: > if (num) {return true}

works as long as "undefined" for falsy values is acceptable. Ternary operator like so:

> num ? true : false

didn't work.

Comment (JohnBain) [inline answers.js]: > return num ? true : false

Figured it out. Of course.

Comment (ziad-saab) [inline answers.js]: yes!!

**Pull request closed without merging**